x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/rank/textsimilarity/TextSimilarityRankBuilderTests.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.rank.textsimilarity;
+
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.test.AbstractXContentSerializingTestCase;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+
+public class TextSimilarityRankBuilderTests extends AbstractXContentSerializingTestCase<TextSimilarityRankBuilder> {
+
+    @Override
+    protected TextSimilarityRankBuilder createTestInstance() {
+        return new TextSimilarityRankBuilder("my-field", "my-inference-id", "my-inference-text", randomIntBetween(1, 1000), randomFloat());
```

Comment:
```suggestion
        return new TextSimilarityRankBuilder("my-field", "my-inference-id", "my-inference-text", randomIntBetween(1, 1000), randomBoolean() ? null : randomFloat());
```
We need to test null handling.